lib/cmps/sourcepoint-frame.ts

```diff
@@ -0,0 +1,100 @@
+import { enableLogs } from "../config";
+import { click, elementExists, wait, waitForElement } from "../rule-executors";
+import { waitFor } from "../utils";
+import AutoConsentCMPBase from "./base";
+
+export default class SourcePoint extends AutoConsentCMPBase {
+
+  ccpaMode = false;
+
+  constructor() {
+    super("Sourcepoint-frame");
+  }
+
+  get hasSelfTest(): boolean {
+    return false; // self-test is done by parent frame
+  }
+
+  get isIntermediate(): boolean {
+    return false;
+  }
+
+  async detectCmp() {
+    if (window.top === window) {
+      // this rule is only for nested frames
+      return false;
+    }
```

Comment:
Can't you just use the `context` here?

Reply:
Initially I implemented runContext only for declarative rules. But I've just pushed a change to do it for class-based rules too.